homeassistant/components/mqtt/models.py

```diff
@@ -44,26 +42,26 @@ class PayloadSentinel(StrEnum):
 PublishPayloadType = str | bytes | int | float | None
 
 
-@attr.s(slots=True, frozen=True)
+@dataclass
 class PublishMessage:
-    """MQTT Message."""
+    """MQTT Message for publishing."""
 
-    topic: str = attr.ib()
-    payload: PublishPayloadType = attr.ib()
-    qos: int = attr.ib()
-    retain: bool = attr.ib()
+    topic: str
+    payload: PublishPayloadType
+    qos: int
+    retain: bool
 
 
-@attr.s(slots=True, frozen=True)
+@dataclass
```

Comment:
```suggestion
@dataclass(slots=True)
```

Reply:
Could keep it frozen as well but its slower, and I don't think there is much risk of anything modifying it anyways

Reply:
The class is only used internally, so I believe it is okay not to freeze